ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -261,6 +261,32 @@ let check_protection_policy ~vmr ~op ~ref_str =
 		[ref_str; Ref.string_of vmr.Db_actions.vM_protection_policy])
 	| _ -> None
 
+(** Some VMs can't migrate. The predicate [check_mobile] is true, if and
+ * only if a VM is mobile.
+ *
+ * A VM is not mobile if in its [last_booted_record] any of the
+ * following values are true: [platform:nomigrate] or
+ * [platform:nested-virt]. If we cannot find a boot record the VM can
+ * migrate. A VM can always migrate if strict=false.
+ **)
+
+let check_mobile strict vm =
+	let is_true str = (* will switch to function in xapi_vm_helpers.ml *)
+		str |> String.lowercase |> function
+		| "true"  | "1" -> true
+		| "false" | "0" -> false
+		| _ -> failwith (Printf.sprintf "can't parse %s as boolean" str)
```

Comment:
It is always better to use '%s' with quotes, which makes the error message easier to read if str contains a space or other non-alpha characters